atlassian_jwt_auth/contrib/django_app/decorators.py

```diff
@@ -0,0 +1,41 @@
+from functools import wraps
+from django.http import HttpResponse, HttpResponseForbidden
+from django.conf import settings
+
+import jwt
+
+from atlassian_jwt_auth.contrib.django_app.auth import ASAPAuthBackend
+
+
+class HttpResponseNotAuthorized(HttpResponse):
+    status_code = 401
+
+
+def asap_required(view_func, issuers=None):
+    """Decorator which ensures the user has provided a correct user and token pair."""
+
+    @wraps(view_func)
+    def _wrapped_view(request, *args, **kwargs):
+        user = request.user
+
+        auth_header = request.META.get('HTTP_AUTHORIZATION', '').split(' ', 1)
+
+        if len(auth_header) < 2 or auth_header[0].lower() != 'bearer':
+            return HttpResponseNotAuthorized('This endpoint requires ASAP authentication.')
+
+        token = auth_header[1]
+        user = ASAPAuthBackend.authenticate(token=token)
+        if user is None:
+            return HttpResponseNotAuthorized('This endpoint requires ASAP authentication.')
+
+        claims = jwt.decode(token, verify=False)
+        if issuers and claims.get('iss') not in issuers:
+            return HttpResponseForbidden('ASAP issuer is not allowed to access this endpoint')
+
+        if not issuers and settings.ASAP_ALLOWED_ISSUERS and claims.get('iss') not in issuers:
+            return HttpResponseForbidden('ASAP issuer is not allowed to access this endpoint')
+
+        request.user = user
```

Comment:
Yes, Django's authentication model requires authentication backends tie credentials to a specific user.